instance/models/load_balancer.py

```diff
@@ -111,6 +148,28 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.domain
 
+    def modifies_configuration(method):  # nopep8 # pylint: disable=no-self-argument
+        """
+        Flag an internal method as one that modifies the state of load balancer configuration,
+        to allow managing configuration state outside and around the method call.
+        """
+        @functools.wraps(method)
+        def wrapper(self, *args, **kwargs):
+            """
+            Used to wrap the function that modified load balancer configuration state.
+            """
+            with transaction.atomic():
+                self.configuration_version += 1
+                self.save()
+
+            with self._configuration_lock(blocking_timeout=1):
+                candidate_configuration_version = self.configuration_version
+                method(self, *args, **kwargs)  # pylint: disable=not-callable
+                self.deployed_configuration_version = candidate_configuration_version
+
+            self.save()
```

Comment:
I think you should save while still holding the lock.

Reply:
Oh woops, good point. I had out of the lock's block because I thought I was going to be adding some extra logic, but I ended up not doing that.